src/Controller/BackendController.php

```diff
@@ -238,4 +242,47 @@ public function pickerAction(Request $request): RedirectResponse
 
         return new RedirectResponse($picker->getCurrentUrl());
     }
+
+    /**
+     * @return Response
+     *
+     * @Route("/contao/2fa", name="contao_backend_2fa")
+     */
+    public function twoFactorAuthAction(): Response
+    {
+        /** @var RouterInterface $router */
+        $router = $this->get('router');
+
+        $this->get('contao.framework')->initialize();
+
+        $token = $this->get('security.token_storage')->getToken();
+
+        if (!$token instanceof TwoFactorToken) {
+            return new RedirectResponse($router->generate('contao_backend_login'));
+        }
+
+        $authenticatedToken = $token->getAuthenticatedToken();
+
+        if (!$authenticatedToken instanceof UsernamePasswordToken) {
+            return new RedirectResponse($router->generate('contao_backend_login'));
+        }
+
+        $user = $authenticatedToken->getUser();
+
+        if (!$user instanceof BackendUser) {
+            return new RedirectResponse($router->generate('contao_backend_login'));
+        }
+
+        return $this->loginAction();
+    }
+
+    /**
+     * Dummy action for symfony security
+     * See https://symfony.com/doc/current/reference/configuration/security.html#check-path.
+     *
+     * @Route("/contao/2fa_check", name="contao_backend_2fa_check")
```

Comment:
are you sure this route is needed? We currently never use check routes, we just use the regular route and it will match for POST requests…

Reply:
I could not get it to work without. See [here](https://symfony.com/doc/current/reference/configuration/security.html#check-path) and [here](https://github.com/scheb/two-factor-bundle/blob/master/Resources/doc/configuration.md#configuration).

Reply:
I even don't get it how it works without a check route for the current security.

Reply:
https://github.com/symfony/symfony-docs/issues/6126

BTW, that someone at SymfonyCon was me 😂

Reply:
Then I get `ERR_TOO_MANY_REDIRECTS`.

Reply:
It's required because of the two-factor bundle. See scheb/two-factor-bundle#145, once it gets fixed we can remove this.